packages/opentelemetry-test-utils/src/instrumentations/otel-provider-api.ts

```diff
@@ -18,27 +18,88 @@ import {
   ReadableSpan,
 } from '@opentelemetry/sdk-trace-base';
 
-const OTEL_TESTING_MEMORY_EXPORTER = Symbol.for(
-  'opentelemetry.testing.memory_exporter'
+import {
+  InMemoryMetricExporter,
+  ResourceMetrics,
+} from '@opentelemetry/sdk-metrics';
+import { TestMetricReader } from '../TestMetricReader';
+
+const OTEL_TRACING_TESTING_MEMORY_EXPORTER = Symbol.for(
+  'opentelemetry.tracing.testing.memory_exporter'
 );
 
-type OTelProviderApiGlobal = {
-  [OTEL_TESTING_MEMORY_EXPORTER]?: InMemorySpanExporter;
+const OTEL_METRICS_TESTING_MEMORY_EXPORTER = Symbol.for(
+  'opentelemetry.metrics.testing.memory_exporter'
+);
+
+const OTEL_METRICS_TESTING_READER = Symbol.for(
+  'opentelemetry.metrics.testing.reader'
+);
+
+type OTelExportersApiGlobal = {
+  [OTEL_TRACING_TESTING_MEMORY_EXPORTER]?: InMemorySpanExporter;
+  [OTEL_METRICS_TESTING_MEMORY_EXPORTER]?: InMemoryMetricExporter;
+  [OTEL_METRICS_TESTING_READER]?: TestMetricReader;
+};
+
+const _global = global as OTelExportersApiGlobal;
+
+export const getTracingTestMemoryExporter = ():
+  | InMemorySpanExporter
+  | undefined => {
+  return _global[OTEL_TRACING_TESTING_MEMORY_EXPORTER];
 };
-const _global = global as OTelProviderApiGlobal;
 
-export const getTestMemoryExporter = (): InMemorySpanExporter | undefined => {
-  return _global[OTEL_TESTING_MEMORY_EXPORTER];
+export const getMetricsTestMemoryExporter = ():
+  | InMemoryMetricExporter
+  | undefined => {
+  return _global[OTEL_METRICS_TESTING_MEMORY_EXPORTER];
 };
 
-export const setTestMemoryExporter = (memoryExporter: InMemorySpanExporter) => {
-  _global[OTEL_TESTING_MEMORY_EXPORTER] = memoryExporter;
+export const getMetricsTestReader = (): TestMetricReader | undefined => {
+  return _global[OTEL_METRICS_TESTING_READER];
+};
+
+export const setMetricsTestMemoryExporter = (
+  memoryExporter: InMemoryMetricExporter
+) => {
+  _global[OTEL_METRICS_TESTING_MEMORY_EXPORTER] = memoryExporter;
+};
+
+export const setTracingTestMemoryExporter = (
```

Comment:
Not sure if we have naming guidance on this, but I'd suggest that we keep the term aligned with the SDK names

```suggestion
export const setTestMemorySpanExporter = (
```